contracts/periphery/DepositPeriphery.sol

```diff
@@ -0,0 +1,156 @@
+    IRewardRouterV2 internal rewardRouter;
+
+    // JIT router for deposits
+    IDnGmxRouter internal jitRouter;
+    // delta neutral junior tranche
+    IDnGmxJuniorVault internal dnGmxJuniorVault;
+
+    /// @notice sets the maximum slippage threshold to be used for converting glp for asset
+    /// @param _slippageThreshold slippage threshold value in bps
+    function setSlippageThreshold(uint256 _slippageThreshold) external onlyOwner {
+        slippageThreshold = _slippageThreshold;
+        emit SlippageThresholdUpdated(_slippageThreshold);
+    }
+
+    /// @notice sets the required external contract address in order to swap glp for tokens
+    /// @dev only owner call this setter function
+    /// @param _jitRouter just-in-time liqudity router for better slippages for junior vault
+    /// @param _dnGmxJuniorVault junior tranche of delta neutral vault
+    /// @param _rewardRouter reward router v2 of gmx protocol
+    function setAddresses(
+        IDnGmxRouter _jitRouter,
+        IDnGmxJuniorVault _dnGmxJuniorVault,
+        IRewardRouterV2 _rewardRouter
+    ) external onlyOwner {
+        jitRouter = _jitRouter;
+        rewardRouter = _rewardRouter;
+        dnGmxJuniorVault = _dnGmxJuniorVault;
+
+        // query sGlp direclty from junior tranche
+        sGlp = ISglpExtended(dnGmxJuniorVault.asset());
+
+        // query glpManager from sGlp
+        glpManager = IGlpManager(sGlp.glpManager());
+
+        // query gmxVault from glpManager
+        gmxVault = IVault(glpManager.vault());
+
+        // give allowance to JIT router
+        sGlp.approve(address(_jitRouter), type(uint256).max);
+        // give allowance to glpManager to pull & burn sGlp
+        sGlp.approve(address(_dnGmxJuniorVault), type(uint256).max);
+
+        emit AddressesUpdated(address(_jitRouter), address(_dnGmxJuniorVault), address(_rewardRouter));
+    }
+
+    /// @notice allows to use tokens to deposit into junior vault
+    /// @param token input token
+    /// @param receiver address of the receiver
+    /// @param tokenAmount amount of token to deposit
+    /// @return sharesReceived shares received in exchange of token
+    function depositToken(
+        address token,
+        address receiver,
+        uint256 tokenAmount
+    ) external returns (uint256 sharesReceived) {
+        IERC20(token).transferFrom(msg.sender, address(this), tokenAmount);
+
+        uint256 glpReceived = _convertToSglp(token, tokenAmount);
+
+        sharesReceived = dnGmxJuniorVault.deposit(glpReceived, receiver);
+
+        emit TokenDeposited(msg.sender, receiver, token, glpReceived, sharesReceived, tokenAmount);
+    }
+
+    /// @notice allows to use tokens to deposit into junior vault using JIT router
+    /// @param token input token
+    /// @param receiver address of the receiver
+    /// @param tokenAmount amount of token to deposit
+    /// @return sharesReceived shares received in exchange of token
+    function depositTokenWithJIT(
```

Comment:
we should add a depositToken in JIT router rather than this to follow common end point for deposit in frontend